client/editSchedule.js

```diff
@@ -43,31 +33,20 @@ setupCompetitionCalendar = function(template, $calendarDiv, $editModal) {
       var duration = moment.duration(timeMinutes, 'minutes');
       return Math.floor(duration.asHours()) + ":" + duration.minutes() + ":00";
     }
-    var calendarStartMinutes = getCompetitionCalendarStartMinutes(competitionId);
-    var calendarEndMinutes = getCompetitionCalendarEndMinutes(competitionId);
+    var compStartMinutes = getCompetitionCalendarStartMinutes(competitionId);
+    var compEndMinutes = getCompetitionCalendarEndMinutes(competitionId);
     var numberOfDays = getCompetitionNumberOfDays(competitionId);
 
-    var startDateMoment = getCompetitionStartDateMoment(competitionId);
-    if(!startDateMoment) {
-      startDateMoment = moment.utc().startOf('day');
-    }
-    var minTime = timeMinutesToFullCalendarTime(calendarStartMinutes);
-    var maxTime = timeMinutesToFullCalendarTime(calendarEndMinutes);
-
-    var eventChanged = function(calEvent) {
-      var nthDay = calEvent.start.diff(startDateMoment, 'days');
-      var startMinutes = calEvent.start.hour()*60 + calEvent.start.minute();
+    var startDateMoment = getCompetitionStartDateMoment(competitionId) || moment.utc().startOf('day');
 
-      var $set = {
-        nthDay: nthDay,
-        startMinutes: startMinutes,
+    var eventChanged = function(event) {
+      var update = {
+        nthDay: event.start.diff(startDateMoment, 'days'),
+        startMinutes: event.start.hour()*60 + event.start.minute(),
+        durationMinutes: event.end.diff(event.start, 'minutes'),
```

Comment:
Just comparing with the code on the left, are we sure that event.end is always defined? The code on the left doesn't seem to think so.